### DISCUSSION TITLE
How does duckdb support for Apache Arrow RecordBatchReaders work under the hood?

### DISCUSSION THREAD

**Discussion by snowch** [duckdb/duckdb#15384] [Q&A] [answered]
There is documentation on the Apache Arrow RecordBatchReaders [here](https://duckdb.org/docs/guides/python/sql_on_arrow.html#apache-arrow-recordbatchreaders).

Is there any documentation how DuckDB works under the hood with a RecordBatchReader?

**Comment by Tishj** [comment, marked as answer]
Could be useful to look at:
`tools/pythonpkg/src/arrow/arrow_array_stream.cpp`

Creates the arrow scan from the python object
file: `tools/pythonpkg/src/python_replacement_scan.cpp` method: `CreateArrowScan`
